cpp/tests/structure/renumbering_test.cpp

```diff
@@ -0,0 +1,176 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governin_from_mtxg permissions and
+ * limitations under the License.
+ */
+
+#include <utilities/high_res_clock.h>
+#include <utilities/base_fixture.hpp>
+#include <utilities/test_graphs.hpp>
+#include <utilities/test_utilities.hpp>
+#include <utilities/thrust_wrapper.hpp>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/graph.hpp>
+#include <cugraph/graph_functions.hpp>
+#include <cugraph/graph_view.hpp>
+
+#include <raft/cudart_utils.h>
+#include <raft/handle.hpp>
+#include <rmm/device_uvector.hpp>
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <iterator>
+#include <limits>
+#include <numeric>
+#include <random>
+#include <vector>
+
+// do the perf measurements
+// enabled by command line parameter s'--perf'
+//
+static int PERF = 0;
+
+struct Renumbering_Usecase {
+  bool check_correctness{true};
+};
+
+template <typename input_usecase_t>
+class Tests_Renumbering
+  : public ::testing::TestWithParam<std::tuple<Renumbering_Usecase, input_usecase_t>> {
+ public:
+  Tests_Renumbering() {}
+  static void SetupTestCase() {}
+  static void TearDownTestCase() {}
+
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  template <typename vertex_t, typename edge_t, typename weight_t>
+  void run_current_test(Renumbering_Usecase const& renumbering_usecase,
+                        input_usecase_t const& input_usecase)
+  {
+    raft::handle_t handle{};
+    HighResClock hr_clock{};
+
+    std::vector<vertex_t> h_original_src_v{};
+    std::vector<vertex_t> h_original_dst_v{};
+    std::vector<vertex_t> h_final_src_v{};
+    std::vector<vertex_t> h_final_dst_v{};
+
+    rmm::device_uvector<vertex_t> src_v(0, handle.get_stream_view());
+    rmm::device_uvector<vertex_t> dst_v(0, handle.get_stream_view());
+    rmm::device_uvector<vertex_t> renumber_map_labels_v(0, handle.get_stream_view());
+    vertex_t number_of_vertices{};
+
+    std::tie(src_v, dst_v, std::ignore, std::ignore, number_of_vertices, std::ignore) =
+      input_usecase.template construct_edgelist<vertex_t, edge_t, weight_t, false, false>(handle,
+                                                                                          false);
+
+    if (renumbering_usecase.check_correctness) {
+      h_original_src_v.resize(src_v.size());
+      h_original_dst_v.resize(dst_v.size());
+
+      raft::update_host(h_original_src_v.data(), src_v.data(), src_v.size(), handle.get_stream());
+      raft::update_host(h_original_dst_v.data(), dst_v.data(), dst_v.size(), handle.get_stream());
+    }
+
+    if (PERF) {
+      handle.get_stream_view().synchronize();  // for consistent performance measurement
```

Comment:
So, we're using CUDA_TRY(cudaDeviceSynchronize()) elsewhere.
This will be lower overhead and proper if every operation is executed on ```stream``` (this should be true... but may not be always true...) but we may better not mix the two in our tests. So, do you think we'd better use stream synchronize in our tests or we'd better continue using cudaDeviceSynchronize?

Reply:
I switched to `CUDA_TRY(cudaDeviceSynchronize())` in the next push.

I'm not particularly concerned about overhead in tests.  We can consider that change as a general possibility later.

Reply:
Yeah...  me, neither. Just wasn't happy about mixing cudaDeviceSynchronize() and stream_view.synchronize(). Either is fine for me as long as we're consistent.